packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -593,77 +642,74 @@ export const addAccountMetadata =
 /**
  * Generate device master-key
  * */
-export const setDeviceMetadataKey = () => async (dispatch: Dispatch, getState: GetState) => {
-    if (!getState().metadata.enabled) return;
-    const { device } = getState().suite;
-    if (!device || !device.state || !device.connected) return;
-
-    // mater key already exists
-    if (device.metadata.status === 'enabled') return;
-
-    const result = await TrezorConnect.cipherKeyValue({
-        device: {
-            path: device.path,
-            state: device.state,
-            instance: device.instance,
-        },
-        useEmptyPassphrase: device.useEmptyPassphrase,
-        path: METADATA.ENABLE_LABELING_PATH,
-        key: METADATA.ENABLE_LABELING_KEY,
-        value: METADATA.ENABLE_LABELING_VALUE,
-        encrypt: true,
-        askOnEncrypt: true,
-        askOnDecrypt: true,
-    });
-
-    if (result.success) {
-        if (!getState().metadata.enabled) {
-            dispatch({
-                type: METADATA.ENABLE,
-            });
-        }
+export const setDeviceMetadataKey =
+    (encodingVersion: number, overwriteExisting = false) =>
```

Comment:
maybe we could introduce to metadataTypes.ts
```
type EncodingVersion = 1 | 2 
```
In metadata constants this type coudl be then stricter 
```

export const CIPHER_KEY_VALUE_CONFIGS: Record<
    MetadataTypes,
    Parameters<TrezorConnect['cipherKeyValue']>[0]['bundle'][0]
> = {
    1: {
       ...    },
    2: {
        ...
    },
};

```

Reply:
done